crates/project/src/project.rs

```diff
@@ -858,6 +865,13 @@ impl Project {
         .await
     }
 
+    fn new_search_history() -> SearchHistory {
```

Comment:
NIT: could be inlined, we use it once only?

Reply:
It's actually used twice inside `Project::from_join_project_response` and `Project::local